### ISSUE TITLE
Clickhouse models that ingest data from external sources using clickhouse SQL should materialize to true by default

### ISSUE THREAD

**Issue by k-anshul** [rilldata/rill#6278]
We can only mark models materialized by default for cases when input and output connector are different. 
But for clickhouse since we rely on clickhouse connectors to ingest all data it is not possible for us to set that default.
This leads to very hard to know issues where clickhouse keeps pulling data from external source for every query.

The same issue will also be present in duckDB when we move to its internal connectors.

**Comment by begelundmuller** [comment]
I wonder if there's a way we can analyze a query for external dependencies? For example in DuckDB, maybe creating a temporary view with `SET enable_external_access=false`? 

> The same issue will also be present in duckDB when we move to its internal connectors.

For backwards compatibility at least, this can be solved by having `type: source` emit a model where `materialize: true`.

**Comment by k-anshul** [comment]
>For example in DuckDB, maybe creating a temporary view with SET enable_external_access=false

Yes this can be evaluated for duckdb. Though `enable_external_access` once set to false can't be set to true and is a global configuration. We may need to do throw away the write handle once we do this check.

**Comment by begelundmuller** [comment]
Closing since this is probably best addressed with this behavioral change: https://www.notion.so/rilldata/Model-syntax-for-sources-196ba33c8f57801fbef1f8a307f76466
